### PR TITLE
Example that creates a NL file from a Pyomo model and use the generated NL file to obtain a `NLPModel`

### PR DESCRIPTION
**Pull request by syarra**

### CODE REVIEW

**Comment by dpo** [comment]
Thanks. Are you able to pass the model to a solver?

**Comment by syarra** [comment]
I updated the examples and one problem is solved using TRON.

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ notifications:
 before_install:
   - brew update
   - brew tap homebrew/science
-  - brew uninstall -f adol-c boost cppad asl metis
+  - brew uninstall --ignore-dependencies adol-c boost cppad asl metis
   - brew install adol-c
```
Comment (dpo) [inline .travis.yml]: You really need `-f` here.

File: .travis.yml [hunk 2]
```diff
@@ -48,3 +48,4 @@ script:
 
 after_success:
   - coveralls
+  -
```
Comment (dpo) [inline .travis.yml]: What's this?

File: examples/demo_pyomo.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    print '\n\n\n'
+
+if __name__ == "__main__":
+    pyomo_simple_model = create_simple_model()
+    nl_filename = "simple.nl"
+    write_nl(pyomo_simple_model, nl_filename)
+
+    # Create an Amplmodel from the generated nl file
+    simple_model = AmplModel(nl_filename)
+    display_model(simple_model)
```
Comment (dpo) [inline examples/demo_pyomo.py]: It seems much simpler to just call [`display_basic_info()`](https://github.com/PythonOptimizers/NLP.py/blob/develop/nlp/model/nlpmodel.py#L687).

File: examples/demo_pyomo.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+    
+    pyomo_rosenbrock_model = create_rosenbrock_model()
+    nl_filename = "rosenbrock.nl"
+    write_nl(pyomo_rosenbrock_model, nl_filename)
+
+    # Create an Amplmodel from the generated nl file
+    rosenbrock_model = AmplModel(nl_filename)
+    display_model(rosenbrock_model)
```
Comment (dpo) [inline examples/demo_pyomo.py]: Let use just one model instead of two. It doesn't add anything new.

File: nlp/optimize/tron.py [hunk 5]
```diff
@@ -426,7 +426,7 @@ def solve(self):
                 self.g_old = self.g.copy()
 
             # Wrap Hessian into an operator.
-            H = model.hop(self.x.copy(), self.model.pi)
+            H = model.hop(self.x.copy())
```
Comment (dpo) [inline nlp/optimize/tron.py]: Does this wreck AugLag?!

File: examples/demo_pyomo.py [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    model.y = Var(bounds=(-1.5, None))
+    model.o = Objective(expr=(model.x-1)**2 + \
+                              100*(model.y-model.x**2)**2)
+    model.x.set_value(-2.)
+    model.y.set_value(1.)
+    return model
+
+def write_nl(model, nl_filename):
+    """Writes a Pyomo model in NL file format."""
+    _, smap_id = model.write(nl_filename,
```
Comment (dpo) [inline examples/demo_pyomo.py]: The `write_nl()` function doesn't really simplify anything. I suggest to remove it and call `model.write()` directly.

File: examples/demo_pyomo.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+tron_logger = config_logger("nlp.tron",
+                            "%(name)-8s %(levelname)-5s %(message)s",
+                            level=logging.INFO)
+
+def create_simple_model():
+    model = ConcreteModel()
+    model.x = Var(bounds=(1., None))
+    model.o = Objective(expr=model.x)
+    model.x.set_value(20.0)
+    return model
```
Comment (dpo) [inline examples/demo_pyomo.py]: Let's remove this model.

**Review by dpo — CHANGES_REQUESTED** [review]

**Review by dpo — APPROVED** [review]

**Pull request merged**